test/OrderByTest.cpp

```diff
@@ -235,3 +236,33 @@ TEST(OrderBy, simpleMemberFunctions) {
     EXPECT_EQ(84.0, s.getMultiplicity(1));
   }
 }
+
+TEST(OrderBy, verifyOperationIsPreemptivelyAbortedWithNoRemainingTime) {
+  VectorTable input;
+  // Make sure the estimator estimates a couple of ms to sort this
+  for (int64_t i = 0; i < 1000; i++) {
+    input.push_back({0, i});
+  }
+  auto inputTable = makeIdTableFromVector(input, &Id::makeFromInt);
+  OrderBy orderBy = makeOrderBy(std::move(inputTable), {{1, false}, {0, true}});
+  // Safe to do, because we know the underlying estimator is mutable
+  const_cast<SortPerformanceEstimator&>(
+      orderBy.getExecutionContext()->getSortPerformanceEstimator())
+      .computeEstimatesExpensively(
+          ad_utility::makeUnlimitedAllocator<ValueId>(),
+          std::numeric_limits<size_t>::max());
```

Comment:
This typically runs for some time, maybe constrain the sort performance estimator (do not pass `max()` etc.)
(same for the `Sort.h`).